infoblox/resource_infoblox_ipv6_network.go

```diff
@@ -0,0 +1,186 @@
+	return resourceIPv6NetworkRead(d, m)
+}
+
+func resourceIPv6NetworkRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Reading the required network block", resourceIPv6NetworkIDString(d))
+
+	networkViewName := d.Get("network_view_name").(string)
+	tenantID := d.Get("tenant_id").(string)
+	connector := m.(*ibclient.Connector)
+
+	objMgr := ibclient.NewObjectManager(connector, "Terraform", tenantID)
+	obj, err := objMgr.GetIPv6NetworkWithRef(d.Id())
+	if err != nil {
+		return fmt.Errorf("Getting Network block from network view (%s) failed : %s", networkViewName, err)
+	}
+	d.SetId(obj.Ref)
+	log.Printf("[DEBUG] %s: Completed reading network block", resourceNetworkIDString(d))
+	return nil
+}
+
+func resourceIPv6NetworkUpdate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Updating IPv6 network block EAs", resourceIPv6NetworkIDString(d))
+
+	networkViewName := d.Get("network_view_name").(string)
+	tenantID := d.Get("tenant_id").(string)
+	connector := m.(*ibclient.Connector)
+	extensibleAttributes := d.Get("extensible_attributes").(map[string]interface{})
+
+	objMgr := ibclient.NewObjectManager(connector, "Terraform", tenantID)
+
+	addEA := make(map[string]interface{})
+	removeEA := make(map[string]interface{})
+
+	var err error
+	var IPv6Network *ibclient.IPv6Network
+	for key, value := range extensibleAttributes {
+		if value == "nil" {
+			removeEA[key] = value
+			IPv6Network, err = objMgr.UpdateIPv6NetworkEA(d.Id(), addEA, removeEA)
+			if err != nil {
+				return fmt.Errorf("Deletion of IPv6 network EA failed from network view(%s): %s", networkViewName, err)
+			}
+		} else {
+			addEA[key] = value
+			IPv6Network, err = objMgr.UpdateIPv6NetworkEA(d.Id(), addEA, removeEA)
+			if err != nil {
+				return fmt.Errorf("Addition of IPv6 network EA failed from network view(%s): %s", networkViewName, err)
+			}
+
+		}
+	}
+	d.SetId(IPv6Network.Ref)
+
+	log.Printf("[DEBUG] %s: Updation on IPv6 network block complete", resourceIPv6NetworkIDString(d))
+	return resourceIPv6NetworkRead(d, m)
+}
+
+func resourceIPv6NetworkDelete(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning Deletion of IPv6 network block", resourceIPv6NetworkIDString(d))
+
+	networkViewName := d.Get("network_view_name").(string)
+	tenantID := d.Get("tenant_id").(string)
+	connector := m.(*ibclient.Connector)
+
+	objMgr := ibclient.NewObjectManager(connector, "Terraform", tenantID)
+	_, err := objMgr.DeleteIPv6Network(d.Id(), d.Get("network_view_name").(string))
+	if err != nil {
+		return fmt.Errorf("Deletion of IPv6 Network block failed from network view(%s): %s", networkViewName, err)
+	}
+	d.SetId("")
+
+	log.Printf("[DEBUG] %s: Deletion of IPv6 network block complete", resourceIPv6NetworkIDString(d))
+	return nil
+}
+
+type resourceIPv6NetworkIDStringInterface interface {
+	Id() string
+}
+
+func resourceIPv6NetworkIDString(d resourceIPv6NetworkIDStringInterface) string {
```

Comment:
I see that you use this function just for debug output. If the debug code is to be removed, do not forget to remove this function as well.